tests/test_crawlers.py

```diff
@@ -599,3 +601,59 @@ def test_iterating(self):
         with self.assertLogs(self.crawler.LOGGER):
             for i, url in enumerate(self.crawler):
                 self.assertEqual(url, expected_urls[i])
+
+
+class FTPCrawlerTestCase(unittest.TestCase):
+    """Tests for the FTP crawler"""
+
+    def emulate_cwd_of_ftp(self, name):
+        """passes in the case of "", ".." or "folder_name" in order to resemble the behavior of cwd
+        of ftplib. Otherwise (encountering a filename) raise the proper exception """
+        if name not in ["..", "folder_name", ""]:
+            raise ftplib.error_perm
+
+    @mock.patch('geospaas_harvesting.crawlers.ftplib.FTP', autospec=True)
+    def test_ftp_correct_navigation(self, mock_ftp):
+        """ shall categorize the specific file names (based on specific 'fileformat' which is
+        revealed in the configuration file) as well as folder(s) inside the ftp resource """
+        test_crawler = crawlers.FTPCrawler('ftp:///', fileformat='.gz')
+        test_crawler.ftp.nlst.return_value = ['file1.gz', 'folder_name', 'file3.bb', 'file2.gz', ]
+        test_crawler.ftp.cwd = self.emulate_cwd_of_ftp
+        test_crawler.ftp.host = ''
+        test_crawler.ftp.pwd.return_value = ''
```

Comment:
due to the modifications in the crawler, that line is not needed anymore
```suggestion
```

Reply:
Done in the next commit